run/observer.ts

```diff
@@ -1,28 +1,53 @@
 import type { Observer, Resolve } from "../types.ts";
 
-import { shift } from "../deps.ts";
+import { Computation, shift } from "../deps.ts";
 
 export function createObservable<T>() {
   let observers = new Map<Observer<T>, Resolve<T>>();
 
-  return {
+  let observable = {
     notify(value: T) {
-      for (let listener of observers.values()) {
-        listener(value);
+      let handlers = [...observers.values()];
+      for (let handler of handlers) {
+        handler(value);
       }
     },
+    *first(): Computation<T> {
+      return yield* shift<T>(function* (k) {
+        let observer = observable.observe();
+        let value = yield* observer;
+        observer.drop();
+        k(value);
+      });
+    },
```

Comment:
This is just a convenient computation to get the first event from an observable.

1. subscribe
2. receive
3. unsubscribe
4. continue

Reply:
👍